src/awesome/button.rs

```diff
@@ -0,0 +1,343 @@
+use std::fmt::{self, Display, Formatter};
+use std::default::Default;
+use std::rc::Rc;
+use rlua::{self, Table, Lua, UserData, ToLua, Value};
+use super::object::{Object, Objectable};
+use super::signal;
+use super::property::Property;
+use super::class::{self, Class};
+use rustwlc::types::KeyMod;
+use xcb::ffi::xproto::xcb_button_t;
+
+#[derive(Clone, Debug)]
+pub struct ButtonState {
+    button: xcb_button_t,
+    modifiers: KeyMod
+}
+
+#[derive(Clone, Debug)]
+pub struct Button<'lua>(Table<'lua>);
+
+impl <'lua> Button<'lua> {
+    pub fn button(&self) -> rlua::Result<Value<'lua>> {
+        let button = self.0.get::<_, ButtonState>("data")?;
```

Comment:
I don't like accessing the table directly, even in functions like this. Add a auto-implemented function for `Objectable` that takes `&self` and returns the `Table<T>` that exists in the `"data"` field. No boiler plate necessary, and it makes it less error prone.